src/http.cpp

```diff
@@ -544,6 +544,7 @@ Connection::async_connect(std::vector<asio::ip::tcp::endpoint>&& endpoints, Conn
             uint32_t start = 30;
             uint32_t interval = 30;
             uint32_t cnt = 1;
+            this->keepalive_ = start;
```

Comment:
If we can't set number of keep-alive probes on a platform, setting a large interval will cause problems.

Reply:
My proposal was to have the default keepalive value as the default value of keepalive_ instead of as the initial value for start here.
However with the new design this is not needed, keepalive_ could even be removed.